src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -205,6 +206,13 @@ impl StreamTableScan {
             column_ids: upstream_column_ids.clone(),
         };
 
+        let properties = self.ctx().with_options().internal_table_subset(); // TODO: Is this even needed? Why is it needed for simple_agg?
+        let catalog_builder = TableCatalogBuilder::new(properties);
+        let catalog = catalog_builder
+            .build(self.base.distribution().dist_column_indices().into(), 0)
```

Comment:
0 here is the `read_prefix_len_hint`. I'm not quite sure what this means.
Is it "prefix_len" the length of the key segment in the Row?
Or something else.